### ISSUE TITLE
show tab number in tab name

### ISSUE THREAD

**Issue by maiermic** [hrjakobsen/slext#57]
Tabs are switched using `alt + <tab-number>`. It is not easy to see which tab has which number. It could be perceived faster if the tab number is shown in the tab name.

**Comment by hrjakobsen** [comment]
I don't think this is something that I have ever seen in tabbed applications. Can you point to some examples where this has been implemented? I have a feeling that it will clutter the tabs, and not enhance usability.

**Comment by maiermic** [comment]
You are right that this is missing in most tabbed applications, but it is a feature that I always miss and try to add when possible (using extensions or plugins). I guess that's a matter of taste. So it should be optional. Personally, I don't like to count tabs or guess the tab number. I want to find it without thinking to much to stay concentrated on the actual work.

**Comment by maiermic** [comment]
It can be done using

```js
function addGlobalStyle(css) {
  var head, style;
  head = document.getElementsByTagName('head')[0];
  if (!head) { return; }
  style = document.createElement('style');
  style.type = 'text/css';
  style.innerHTML = css;
  head.appendChild(style);
}

function addTabNumbers() {
  let style = '';
  for (let n = 1; n <= 8; ++n) {
    style += `
      .slext-tabs .slext-tabs__tab:nth-child(${n}) .slext-tabs__tab-name::before {
        content: '${n} ';
      }
      `;
  }
  addGlobalStyle(style);
}

addTabNumbers();
```

**Comment by hrjakobsen** [comment]
It would be easier to just add it in the SCSS directly. Adding it is not the difficult part, it is presenting it in a way that is easy to read and doesn't clutter the tabs. The easy solution is of course to just make a toggle that enables/disables the numbers. Other options includes showing the numbers when ALT is pressed, but that may defeat the purpose of showing them at all. 

```diff
diff --git a/app/styles/contentscript.scss b/app/styles/contentscript.scss
index aca72f5..80afe63 100644
--- a/app/styles/contentscript.scss
+++ b/app/styles/contentscript.scss
@@ -571,4 +571,10 @@ header.toolbar.toolbar-tabs {
 
 .file-tree-list {
     padding-bottom: 32px; /* To ensure that the last element is visible if #ide-body overflows page*/
-}
\ No newline at end of file
+}
+
+@for $i from 1 through 9 {
+    .slext-tabs .slext-tabs__tab:nth-child(#{$i}) .slext-tabs__tab-name::before {
+        content: '#{$i} ';
+    }
+}
```

**Comment by maiermic** [comment]
If in doubt, keep it simple :wink:  It is not beautiful, but it is functional and not overly confusing (what's going on). Hence, it is ok for now in my opinion if there is an option in the settings to toggle the visibility of the tab numbers. If tab numbers are hidden by default, it should not be unexpected to anybody what the numbers mean. Style might be improved in a future version.
